### PR TITLE
Feature/main refactoring

### PR DESCRIPTION
**Pull request by rscircus**
Getting into the gory details of main.cpp and general program structure.

### CODE REVIEW

**Comment by rscircus** [comment]
This will be rebased when #1 is in `develop`.

**Review by steffi7574 — APPROVED** [review]

File: src/main.cpp [hunk 1]
```diff
@@ -316,7 +321,7 @@ int main(int argc, char *argv[]) {
  * choosing xdot to be a vector of all ones, ybar = 1.0;
  * ==================================================================================*/
  // TODO: Why is this disabled?
```
Comment (steffi7574) [inline src/main.cpp]: I am pretty sure that the below tests don't work anymore. But somehow I'd want to keep them, as a template, for including them later again...

Comment (rscircus) [reply src/main.cpp]: If you agree, I would park them in another branch and get them out of master.

Comment (steffi7574) [reply src/main.cpp]: Yes I do agree!

**Comment by rscircus** [comment]
This PR will have quite a negative number of added lines of code. :joy:

File: src/main.cpp [hunk 2]
```diff
@@ -201,22 +196,19 @@ int main(int argc, char *argv[]) {
...
-// TODO: If this is excluded nothing works: I suggest a removal of the preprocessor directive
-#if OPTIMIZATION_ENABLED
-
+  /* Measure wall time */
   StartTime = MPI_Wtime();
   StopTime = 0.0;
   UsedTime = 0.0;
 
-
   /* The following loop represents the paper's Algorithm (2) */
```
Comment (steffi7574) [inline src/main.cpp]: /* Main optimization iteration */

Comment (rscircus) [reply src/main.cpp]: Addressed in bd265a3.

**Review by steffi7574 — COMMENTED** [review]
Top

**Pull request merged**